client/src/main/java/org/eclipse/hono/client/HonoConnection.java

```diff
@@ -218,6 +218,15 @@ static HonoConnection newConnection(final Vertx vertx, final ClientConfigPropert
      */
     boolean isShutdown();
 
+    /**
+     * Gets the remote container id as advertised by the peer.
+     * <p>
+     * Returns {@code null} if the client is currently not connected.
```

Comment:
I do not think that we should put that much semantics into the returned value in order to not encourage users to actually test connectivity using this property ...